doc/build-with-vs.md

```diff
@@ -1,77 +1,72 @@
 Build with Visual Studio
 ========================
 
-1. Install a recent version of [CMake][cmake]
-2. Install Visual Studio 14 2015 or Visual Studio 15 2017 with the
-   C/C++ components
-
-3. Prepare a directory structure for DLite and 3rd party libraries:
-
-     1. Select a root folder to create the directory structure. Ex:
-
-            "C:\Users\{username}\Documents\"
-
-     2. Create the following directory:
-
-            {root}\local
-
-     3. Clone the DLite repository in the root folder, creating the folder
-
-            {root}\dlite
-
-4. Download hdf5 library archive from
-   [https://support.hdfgroup.org/ftp/HDF5/current/src/][hdf5]
-
-5. Copy of the file
-
-            {root}\dlite\bootstrap-win.sh
-
-   into the root folder and edit it as follows:
-
-	1. Modify the variable CMAKE_PATH to the path of cmake directory
-	2. Modify the variable ROOT_PATH to your root folder
-	3. Modify the variable PYTHON_EXECUTABLE to Python version 3.x
-	   executable
-	4. Check the version number of hdf5 library
-    5. To build in Win64 mode, add '-G "Visual Studio 15 Win64"' after
-       cmake command when generating the solution (not building)
-
-6. Open a git-bash window:
-	1. Change the directory to your root folder
-	2. Type `sh bootstrap-win.sh` and press enter
-
-7. Open the file
-
-            {root}\dlite\build\dlite.sln
-
-   in Visual Studio
-
-8. In Visual Studio:
-
-    1. Select the solution configuration "Debug", then build the
-       solution (Menu Build -> Build solution)
-    2. Select the solution configuration "Release", then build the
-       solution (Menu Build -> Build solution)
-
-To run the tests, do
-
-    ctest -C Debug
-
-
-Summary to build and install DLite when hdf5 lib are installed
-in the given path LOCAL_DIR
-
-    LOCAL_DIR=/C/Users/tco/Documents/Programs/philib/local
-    PATH=$PATH:$LOCAL_DIR/bin
-    cd dlite
-    mkdir build && cd build
-    cmake -G "Visual Studio 14 2015 Win64" -DHDF5_DIR=$LOCAL_DIR/cmake/hdf5 -DCMAKE_INSTALL_PREFIX=$LOCAL_DIR/ ..
-    cmake --build . --config Debug --target doc
-    cmake --build . --config Debug --target install
-    cmake --build . --config Release --target install
-    ctest -C Debug
-    ctest -C Release
+1. Install Visual Studio 15 2017 or Visual Studio 16 2019 with the
```

Comment:
OK, even better. I will update the documentation accordingly.

Reply:
I like it when tests pass, but in this case it puzzles me why for your Python-tests work. The PATH environment is not set correctly via CMake (See https://github.com/SINTEF/dlite/issues/149#issue-1003926905) - this is because semicolon path-separators are interpreted as a list in CMakes set_property(TEST §{test} ENVIRONMENT "PATH=${dlite_PATH_NATIVE}"). The python package can therefore not find the dlls via the PATH. But where does it find your dlls? The AddDllDirectory workaround in ./bindings/python/__init__.py is not working on my system. Is it working on yours? or do you have dlite.dll and dlite-utils.dll somewhere Windows look for dlls by default (Such as the system-directory?) maybe from a previous buildt?

Reply:
I have no idea why it works for me but not for you. I use git bash together with the below script to configure, compile and run the tests. Maybe the ways dlite handle environment variables plays better with bash than the dos prompt? But I have tried to make sure that all paths confirms to Windows... 

```bash
#!/bin/sh

set -xe

export SOURCE_ROOT="C:\Users\jesperf\source\repos\dlite"
export PYTHON_LIBDIR="C:\Program Files\Python38"


# Compile
export BUILD_ROOT="${SOURCE_ROOT}\build-win64"
mkdir -p "${BUILD_ROOT}"
cd ${BUILD_ROOT}
cmake -G "Visual Studio 15 2017 Win64" \
      -A x64 \
      -DWITH_DOC=OFF \
      -DWITH_HDF5=OFF \
      -DCMAKE_CONFIGURATION_TYPES:STRING="Release" \
      -DCMAKE_INSTALL_PREFIX:PATH="${SOURCE_ROOT}\install\win64" \
      ..

cmake --build . --config Release
cmake --install .
ctest -C Release


# Test with build root
export DLITE_USE_BUILD_ROOT=YES
export PATH="${BUILD_ROOT}\src\Release;${BUILD_ROOT}\src\utils\Release;${PYTHON_LIBDIR}"
export PYTHONPATH="${BUILD_ROOT}\bindings\python"
export DLITE_STORAGES="${SOURCE_ROOT}/src/tests/*.json"

#src/tests/Release/test_storage.exe
python.exe "${SOURCE_ROOT}\bindings\python\tests\test_python_bindings.py"



# Test installed
export INSTALL_PREFIX="${SOURCE_ROOT}\install\win64"

export DLITE_USE_BUILD_ROOT=NO
export DLITE_ROOT="${INSTALL_PREFIX}"
export PATH="${INSTALL_PREFIX}\bin;${PYTHON_LIBDIR}"
export PYTHONPATH="${INSTALL_PREFIX}/lib/site-packages"

#src/tests/Release/test_storage.exe
#python.exe "${SOURCE_ROOT}\bindings\python\tests\test_python_bindings.py"
```

Reply:
I tried this in Git bash. "ctest -C Release" is still failing for tests 41-48 on my system. The dlls cannot be found. Since I am using Anaconda, I might be hitting the following issue: https://github.com/pytorch/pytorch/issues/17051. If that is the case, the AddDllDirectory mechanism in ./bindings/python/\_\_init\_\_.py is not working for me but it may be working for your system with a different Python distribution.

When I set the PATH via export etc. the Python tests pass both in Git bash or from the Windows command prompt.

Summing up, I believe it is best to remove the AddDllDirectory components from \_\_init\_\_.py and set the PATH correctly in CMake (https://github.com/SINTEF/dlite/issues/149) so that the tests can pass. For now, can we close this issue and merge into master?